src/vivarium_cluster_tools/psimulate/branches.py

```diff
@@ -206,34 +209,76 @@ def calculate_keyspace(branches: List[Dict]) -> Dict:
     return keyspace
 
 
-def load_branch_configuration(path: Path) -> Tuple[List[Dict], int, int, Optional[List[int]]]:
+def load_branch_configuration(
+    path: Path,
+) -> Tuple[List[Dict], int, int, Optional[List[int]], Optional[List[int]]]:
     data = yaml.full_load(path.read_text())
 
     input_draw_count = data.get("input_draw_count", 1)
     random_seed_count = data.get("random_seed_count", 1)
     input_draws = data.get("input_draws", None)
-
-    # Validate configuration of input_draws and input_draw_count
-    if "input_draw_count" in data and "input_draws" in data:
-        if len(input_draws) != input_draw_count:
-            raise ValueError(
-                f"Both input_draw_count and input_draws are defined but they are inconsistent. "
-                f"input_draw_count is {input_draw_count} while input_draws has length {len(input_draws)}. "
-            )
-    if input_draws:
-        if [d for d in input_draws if d not in range(0, 1000)]:
-            raise ValueError(
-                f"input_draws contains draws outside of 0-999: {[d for d in input_draws if d not in range(0, 1000)]}"
-            )
-    if input_draw_count < 1 or input_draw_count > 1000:
-        raise ValueError(f"input_draw_count must be within 1-1000. Given: {input_draw_count}")
+    random_seeds = data.get("random_seeds", None)
+
+    # Validate configuration of counts and values for input_draws and random_seeds
+    _check_count_and_values(
+        data, input_draw_count, input_draws, "input_draw_count", "input_draws", 1000
+    )
+    _check_count_and_values(
+        data,
+        random_seed_count,
+        random_seeds,
+        "random_seed_count",
+        "random_seeds",
+        10000,
```

Comment:
Where does this number come from? Just a reasonable max?

Reply:
Good question! For this PR, it comes from the existing seeds implementation. That said, I don't think there is a strong constraint on the max number of seeds or that they must be between 0-9999.